fw/signal/signal.go

```diff
@@ -57,8 +58,8 @@ func (ctx *Context) HandleCancel() {
 			}
 		}
 
-		fmt.Println("Signal received; will wait 10 seconds for cleanup to occur")
-		time.Sleep(10 * time.Second)
+		fmt.Printf("Signal received; will wait %v for cleanup to occur\n", waitTime)
+		time.Sleep(waitTime)
```

Comment:
Is there any way to make this check for success and not wait the whole time?

Reply:
Um there are some tricks we can do with the context to do that; but for now this works and is less than ideal but something that we can use without too much in the heisenbug department.

I'd like to keep it this way for now and maybe make it smoother later, by allowing the user to close a channel to signal when they are done shutting down properly.

Reply:
Let me know if you're comfortable with that.